externs.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * @constructor
+ * @param {(string|!NodeName)} nodeName
+ * @param {Object=} opt_attributes
+ * @param {Array<VNode>=} opt_children
+ */
+function VNode(nodeName, opt_attributes, opt_children) {}
+
+/** @record
+ * @return {?}
+ */
+function NodeName() { }
+
+/** @type {Object} */
+NodeName.prototype.defaultProps
+
+/** @type {(string|!NodeName)} */
+VNode.prototype.nodeName;
+
+/** @type {Object|undefined} */
+VNode.prototype.attributes;
+
+/** @type {Array<VNode>|undefined} */
+VNode.prototypechildren;
```

Comment:
lol nice, I think this did not end up mattering really, since we are not using closure compiler for compilation safety and were only using it to describe the "external" shape for compilation